server/tests/e2e/maps.e2e-spec.ts

```diff
@@ -2396,4 +2458,93 @@ describe('Maps', () => {
                 status: 401
             }));
     });
+
+    describe('GET /api/maps/{mapID}/ranks/{rankNumber}', () => {
+        it('should return the rank info for the rank and map specified', async () => {
+            const res = await get({
+                url: `maps/${map1.id}/ranks/1`,
+                status: 200,
+                token: user1Token
+            });
+
+            expect(res.body.rank).toBe(1);
+            expect(res.body.mapID).toBe(map1.id);
+            expect(res.body.userID).toBe(admin.id);
+            expect(Number.parseInt(res.body.runID)).toBe(Number.parseInt(run1.id));
```

Comment:
This could pass incorrectly for large BigInts, since they lose precision when converted to JSON. For example
`Number.parseInt(BigInt(111111111111111111111))` becomes `111111111111111110000`. So use `BigInt(x)`, not `Number.parseInt(x)`